config/routes.rb

```diff
@@ -0,0 +1,68 @@
+Rails.application.routes.draw do
+  # The priority is based upon order of creation: first created -> highest priority.
+  # See how all your routes lay out with "rake routes".
+
+  # You can have the root of your site routed with "root"
+  root 'welcome#index'
+
+  get '/welcome/markets' => 'welcome#index_markets'
+  get '/welcome/markets/:id' => 'welcome#show_market', as: 'show_market'
+
+  resources :markets do
+    resources :vendors
+  end
+
+  resources :vendors do
+    resources :products
+    resources :sales
+  end
+
+  # Example of regular route:
+  #   get 'products/:id' => 'catalog#view'
```

Comment:
I like to delete these kinds of comments, especially in the routes file. It makes it easier for me to quickly see all the routes without getting distracted by other text.